init_update.lua

```diff
@@ -0,0 +1,183 @@
+  local extracted_dir = join_paths(bootstrap.core_install_dir, name .. "-" .. commit)
+
+  return async(function()
+    if not plug.commit then
+      error("commit missing for plugin: " .. repo)
+    end
+    local _, dir_exists = await(async_stat(extracted_dir))
+    local r = result.ok()
+    -- skip plugins that are already installed
+    if dir_exists then
+      return r
+    end
+
+    local url = "https://github.com/" .. repo .. "/archive/" .. zip_name
+
+    return r
+      :and_then(await, jobs.run({ "curl", "-LO", url }, { cwd = download_dir }))
+      :and_then(
+        await,
+        jobs.run({ "unzip", "-o", join_paths(download_dir, zip_name), "-d", bootstrap.core_install_dir }, {})
+      )
+      :or_else(function()
+        error("download and install failed for plugin '" .. repo .. "'")
+      end)
+  end)
+end
+
+local timer = {}
+function timer:start()
+  self.time = uv.hrtime()
+end
+function timer:stop()
+  return (uv.hrtime() - self.time) * 1e-6
+end
+
+-- prevent packer from loading plugins on run hooks
+local load_plugin = require("packer.plugin_utils").load_plugin
+require("packer.plugin_utils").load_plugin = function() end
+print "Downloading core plugins..."
+
+local packer_stage = 0
+async(function()
+  timer:start()
+  local tasks = {}
+  for _, plug in ipairs(core_plugins) do
+    table.insert(tasks, download_and_install(plug))
+  end
+  a.wait_all(unpack(tasks))
+  await(a.main)
+
+  print("Downloaded core plugins in:", timer:stop(), "ms")
+  vim.fn.delete(download_dir, "rf")
+  vim.fn.delete(download_dir, "d")
+  vim.fn.delete(bootstrap.packer_cache_path)
+  vim.fn.delete(bootstrap.lua_cache_path)
+
+  local plugin_loader = require "lvim.plugin-loader"
+  plugin_loader.load { core_plugins }
+
+  packer.on_complete = function() end
+  packer.on_compile_done = function()
+    if packer_stage == 0 then
+      print("Installed core plugins in:", timer:stop(), "ms")
+      packer_stage = 1
+
+      timer:start()
+      print "Loading core plugins..."
+      packer.compile()
+    elseif packer_stage == 1 then
+      for _, core_plugin in pairs(core_plugins) do
+        pcall(load_plugin, core_plugin)
+      end
+      print("Loaded core plugins in:", timer:stop(), "ms")
+      packer_stage = 2
+
+      require("lvim.config"):init()
+      local user_lvim = get_lvim_after_user_config()
+      print "Installing user plugins..."
+      timer:start()
+
+      for _, entry in ipairs(user_lvim.plugins) do
+        print("-", entry[1])
+      end
+
+      packer.on_complete = packer.on_compile_done
+      require("lvim.plugin-loader").load { core_plugins, user_lvim.plugins }
+      packer.install()
+    elseif packer_stage == 2 then
+      packer_stage = 3
+      packer.compile()
+    elseif packer_stage == 3 then
+      print("Installed user plugins in:", timer:stop(), "ms")
+      vim.cmd [[qall!]]
+    end
+  end
+
+  timer:start()
+  print "Installing core plugins..."
+  packer.sync()
+end)()
```

Comment:
This should all go into `bootstrap.lua` and `plugin-loader` because it's way too complex to manage otherwise.

Reply:
I'm not sure what you mean, this is only for init_update and nothing else. What parts of this do you think should go into bootstrap and plugin-loader respectively?